frontend/src/features/dashboard/components/DeliveryInformationModal.tsx

```diff
@@ -34,30 +46,553 @@ const GridItem = ({ label, value }: { label: string; value: string }) => (
+                                            />
+                                        )}
+                                    />
+                                </Grid>
+                                <Grid item xs={6}>
+                                    <Input
+                                        label="Quantity*"
+                                        type="number"
+                                        value={temporaryDelivery.quantity ?? ''}
+                                        error={formError && !temporaryDelivery?.quantity}
+                                        onChange={(e) =>
+                                            setTemporaryDelivery((curr) => ({
+                                                ...curr,
+                                                quantity: e.target.value ? parseFloat(e.target.value) : undefined,
+                                            }))
+                                        }
+                                    />
+                                </Grid>
+                                <Grid item xs={6}>
+                                    <Autocomplete
+                                        id="uom"
+                                        value={
+                                            temporaryDelivery.measurementUnit
+                                                ? {
+                                                      key: temporaryDelivery.measurementUnit,
+                                                      value: getUnitOfMeasurement(temporaryDelivery.measurementUnit),
+                                                  }
+                                                : null
+                                        }
+                                        options={UNITS_OF_MEASUREMENT}
+                                        getOptionLabel={(option) => option?.value ?? ''}
+                                        renderInput={(params) => (
+                                            <Input
+                                                {...params}
+                                                label="UOM*"
+                                                placeholder="Select unit"
+                                                error={formError && !temporaryDelivery?.measurementUnit}
+                                            />
+                                        )}
+                                        onChange={(_, value) => setTemporaryDelivery((curr) => ({ ...curr, measurementUnit: value?.key }))}
+                                        isOptionEqualToValue={(option, value) => option?.key === value?.key}
+                                    />
+                                </Grid>
+                                <Grid item xs={6}>
+                                    <Input
+                                        id="tracking-number"
+                                        label="Tracking Number*"
```

Comment:
handled with star but following samm model should not be mandatory. If implemented like this in backend, we need to fix that.
Also not validated above (fine), but backend fails without (wrong).